lib/phoenix/live_dashboard/telemetry_listener.ex

```diff
@@ -20,15 +20,23 @@ defmodule Phoenix.LiveDashboard.TelemetryListener do
 
     entries =
       for {metric, index} <- metrics,
-          keep?(metric, metadata),
-          measurement = extract_measurement(metric, measurements) do
-        label = tags_to_label(metric, metadata)
+          map = prepare_entry(metric, measurements, metadata, time) do
+        %{label: label, measurement: measurement, time: time} = map
         {index, label, measurement, time}
       end
 
     send(parent, {:telemetry, entries})
   end
 
+  def prepare_entry(metric, measurements, metadata, time \\ nil) do
+    if keep?(metric, metadata) do
+      time = time || System.system_time(:microsecond)
+      measurement = extract_measurement(metric, measurements)
+      label = tags_to_label(metric, metadata)
+      %{label: label, measurement: measurement, time: time}
+    end
+  end
```

Comment:
Great job! Note to self: I need to make this function a public part of our API. I will do so after merging.